examples/kyb-app/src/domains/collection-flow/helpers/extract-files-metadata.ts

```diff
@@ -0,0 +1,34 @@
+import { isFileId } from '@app/domains/collection-flow/helpers/is-file-id';
+import { DocumentConfiguration, FlowData } from '@app/domains/collection-flow/types';
+import { traverseObjectAndPickValue } from '@app/utils/traverse-object-and-pick-value';
+
+export type FilePropertyKey = string;
+
+export interface WorkflowFileMetadata {
+  fieldName: FilePropertyKey;
+  fileId: string;
+  file: File | null;
+}
+
+export const extractFilesMetadata = (
+  flowData: FlowData,
+  config: DocumentConfiguration[],
+): WorkflowFileMetadata[] => {
+  const filesMetadata = config.reduce((filesPayload, config) => {
+    const value = traverseObjectAndPickValue(config.name, flowData, null) as unknown;
+
+    if (isFileId(value)) {
```

Comment:
What was the need for explicitly testing for CUID here versus `undefined` or `string`?

Reply:
This version inlcudes feature where we can define that document type is an URL.In this case those values should not be processed so in this code im looking only for CUID and using them to download files metadata from /storage